src/shell/focus/order.rs

```diff
@@ -0,0 +1,394 @@
+use std::{ops::ControlFlow, time::Instant};
+
+use cosmic_comp_config::workspace::WorkspaceLayout;
+use keyframe::{ease, functions::EaseInOutCubic};
+use smithay::{
+    desktop::{layer_map_for_output, LayerSurface, PopupKind, PopupManager},
+    output::{Output, OutputNoMode},
+    utils::{Logical, Point},
+    wayland::{session_lock::LockSurface, shell::wlr_layer::Layer},
+    xwayland::X11Surface,
+};
+
+use crate::{
+    backend::render::ElementFilter,
+    shell::{
+        layout::{floating::FloatingLayout, tiling::ANIMATION_DURATION},
+        Shell, Workspace, WorkspaceDelta,
+    },
+    utils::{geometry::*, prelude::OutputExt, quirks::WORKSPACE_OVERVIEW_NAMESPACE},
+    wayland::protocols::workspace::WorkspaceHandle,
+};
+
+pub enum Stage<'a> {
+    SessionLock(Option<&'a LockSurface>),
```

Comment:
Would it make sense to not use an `Option` here, and just never invoke the callback if there is no session lock surface for the output?

At least as things work currently, it looks like that would behave the same. `session_lock_elements` produces no elements if its argument is `None`. `surface_under` and `element_under` should return `None`.

Reply:
I think this is correct, if we have a lock, we are not supposed to render anything else, e.g. in case the client dies.